demo/api.html

```diff
@@ -1,11 +1,212 @@
+                                     //     0         1    2
+parentRule.rule = /(.+):(.*)\//;     // "abc:def/", "abc", ""
+childrenRule.parentGroup = 2;        // Listen for ""
+childrenRule.rule = /(.+)/;          // not match (because the + sign)
+
+// Testing address: http://localhost/abc:def/ghi/
+                                     //     0         1      2      3
+parentRule.rule = /(.+):(.*)\/(.+)/; // "abc:def/", "abc", "def", "ghi/"
+childrenRule.parentGroup = 3;        // Listen for "ghi/"
+childrenRule.rule = /.+/;            // "ghi/"</code></pre>
+        <p>
+          If you're not very familiar with Regular Expressions, <a href="https://regex101.com/#javascript"
+                                                                   target="_blank">https://regex101.com/#javascript</a>
+          is a great tool to test your expressions, so copy the part of URL that you want to match, and others that you
+          don't want match, and create your rule, them you'll be ready to use-it.
+        </p>
+        <p>
+          With PushStateTree you can create a tree with as many levels you need for your project. You can add the
+          rules on fly, even children one's. So you can load entry points of your application with logic data relative
+          to the partial being rendered by your SPA, them you can load the other entry points with your own strategy.
+          <ul>
+            <li>
+              <strong>Pre-cache strategy</strong>: Render the current entry point and start loading other entry points in the background.
+            </li>
+            <li>
+              <strong>On-fly strategy</strong>: It loads the current URL, if the user click in a URL that is out of the current rule, it
+              will download the entry point package, apply the logic and trigger the route.
+            </li>
+          </ul>
+          Those are two examples that you can implement in your SPA using PushStateTree. A good way to archive great
+          results it's to adopt <a href="https://webpack.github.io/" target="_blank">Webpack Module Bundler</a> to pack
+          your bundle in entry points.
+        </p>
+        <p>
+          When you set a <code>parentGroup</code> you can have as many as siblings watching the same group or
+          different groups. It's a good practice to have one instance to watch a group and every rule that apply to such
+          group be children of the rule.
+        </p>
+        <pre class="prettyprint lang-html"><xmp><pushstatetree-rule id="myRule" rule="/page/(.*)/">
+  <pushstatetree-rule id="subpageRule" rule="/(subpageA|subpageB)\/(.*)/" parentGroup="1">
+    <pushstatetree-rule id="numberRule" rule="/([0-9]*)/" parentGroup="2">
+
+    </pushstatetree-rule>
+    <pushstatetree-rule id="helpPluginA" rule="/(.+)/">
+
+    </pushstatetree-rule>
+  </pushstatetree-rule>
+  <pushstatetree-rule id="helpPluginB" rule="/(.+)/" parentGroup="1">
+
+  </pushstatetree-rule>
```

Comment:
it'd be nice to have concrete examples rather than abstract ones. For instance, `/users/123/foo`, `/help/how-do-i-create-a-topic/`, `/modules/emails/email/123/`, /modules/forums/topic/1/` etc.

In this example you are referencing the parentGroup, but are doing nothing with them, so the example itself isn't helping much